### DISCUSSION TITLE
Monotype or dual types

### DISCUSSION THREAD

**Discussion by lostcat420** [PokeAPI/pokeapi#960] [Q&A] [answered]
If wanted to query for Pokemon that are monotypes or a combination of two types, what is the best way to do this? 

I was thinking of the idea to find dual types by querying both types individually, then checking for duplicates. However, if I wanted to find monotypes, it's not possible to exclude Pokemon that are dual types. 

I'm fairly unsophisticated when it comes to this type of stuff, so apologies in advance  😅

**Comment by Naramsim** [comment, marked as answer]
Hi, the best is to use GraphQL:

```gql
query samplePokeAPIquery {
  pokemon_v2_pokemon(where: {pokemon_v2_pokemontypes_aggregate: {count: {predicate: {_gte: 2}}}}) {
    id
    pokemon_v2_pokemontypes {
      pokemon_v2_type {
        name
      }
    }
    name
  }
}
```

https://pokeapi.co/docs/graphql

The query extracts all the pokemon with 2 or more types.

**Reply by lostcat420** [reply]
Thank you! That's been helpful to let me reverse engineer a little bit. I was playing around with the console and trying to write some queries, but having difficulty. Ideally, I'd like to query and return something like:

get dual type combinations:
```
- pokemonTypeA
-- pokemonTypeB
--- pokemonName
--- pokemonName
-- pokemonTypeC
--- pokemonName
```


get monotypes:
```
- pokemonTypeA
-- pokemonName
-- pokemonName
- pokemonTypeB
-- pokemonName
-- pokemonName
- pokemonTypeC
-- pokemonName
-- pokemonName
```
I tried something like this, but not getting good results.. it seems like I can't query the Pokemon names nested underneath the Pokemon Types? 
```
query monotype {
  pokemon_v2_type {
    name
    pokemon_v2_pokemontypes(where: {pokemon_v2_type: {pokemon_v2_pokemontypes_aggregate: {count: {predicate: {_eq: 1}}}}}) {
      id
      pokemon_v2_pokemon {
        name
      }
    }
  }
}
```

Perhaps I am misusing GraphQL here? I should be trying to query the Pokemon then use a different tool to reformat in the way I need?
